packages/vite/src/node/optimizer/index.ts

```diff
@@ -775,9 +775,11 @@ async function prepareEsbuildOptimizerRun(
   if (optimizerContext.cancelled) return { context: undefined, idToExports }
 
   const define = {
-    'process.env.NODE_ENV': JSON.stringify(
-      process.env.NODE_ENV || environment.config.mode,
-    ),
+    'process.env.NODE_ENV': environment.config.keepProcessEnv
+      ? // define process.env.NODE_ENV even for keepProcessEnv === true
+        // as esbuild will replace it automatically when `platform` is `'browser'`
+        'process.env.NODE_ENV'
```

Comment:
```suggestion
    'process.env.NODE_ENV': environment.config.keepProcessEnv
      ? 'process.env.NODE_ENV'
```
is the comment really helpful? 🤔

Reply:
I see, yeah that's not obvious 😅 

I actually didn't understand this before, I thought that your comment was practically saying that `process.env.NODE_ENV` __can__ be defined even for keepProcessEnv === true (since esbuild just replaces it) and not that it __needs__ to